### PR TITLE
Fix Lesson TOC Issue With Intl Characters

### LINKED ISSUE #111: Table of contents fails on international characters
Detected on https://rocketvalidator.com/s/4ec0b390-f775-4db1-87a4-ced8f5ece619/p/4686420/i

It looks like the tables of contents for lessons does not generate the headings when those contain non-English characters.

While in the legacy site it works fine:

https://elixirschool.com/es/lessons/specifics/mnesia/#inicialización-de-datos-y-migración

the beta site doesn't work OK:

https://beta.elixirschool.com/es/storage/mnesia

Captura de pantalla 2021-07-29 a las 13 22 03
  
Headings containing non-English characters like "inicialización" are not being generated.

### PR DESCRIPTION
**Pull request by kinson**
# Overview

Resolves #111 

Updated the regex used to match headings inside lessons such that accented and international characters are not excluded.

Screen Shot 2021-08-06 at 6 58 51 PM

### CODE REVIEW

**Comment by kinson** [comment]
This deployment depends on https://github.com/elixirschool/elixirschool/pull/2622 being merged to fix a regex exception thrown by the merge conflict.

**Review by doomspork — APPROVED** [review]
Sweet! This is an awesome fix 🎉

**Review by jaimeiniesta — CHANGES_REQUESTED** [review]
Much better! This now fixes the TOC generation, all the sections are included now, but it something is broken, the generated IDs have malformed characters, see:

https://school-house-pr-122.herokuapp.com/es/basics/documentation#documentaci�n-en-linea

The W3C validator reports malformed characters:

https://validator.w3.org/nu/?doc=https%3A%2F%2Fschool-house-pr-122.herokuapp.com%2Fes%2Fbasics%2Fdocumentation

It's weird because the content itself is OK, only the IDs are malformed.

**Comment by doomspork** [comment]
There may very well be a better way for us to handle this stuff. What is here is very much my MVP 😁  I will take a peek at this today as well @kinson 👍

**Comment by kinson** [comment]
> There may very well be a better way for us to handle this stuff. What is here is very much my MVP 😁 I will take a peek at this today as well @kinson 👍

I'm taking a look at this now and may have figured out the issue, deploying now to hopefully resolve this issue 🤞🏼

File: lib/school_house/content/lesson.ex [hunk 1]
```diff
@@ -62,7 +62,7 @@ defmodule SchoolHouse.Content.Lesson do
       |> String.downcase()
 
     name
-    |> String.replace(~r/[^\w\s]/, "")
+    |> String.replace(~r/[^\p{L}\s]/u, "")
```
Comment (kinson) [inline lib/school_house/content/lesson.ex:65]: @jaimeiniesta good catch on the ids! I updated this regex to handle the unicode characters and use the same `\p{L}` regex, now the links appear to be correct e.g. https://school-house-pr-122.herokuapp.com/es/basics/documentation#documentaci%C3%B3n-en-linea

Comment (jaimeiniesta) [reply lib/school_house/content/lesson.ex:65]: Cool! I just checked and it's fixed now. W3C Validator doesn't complain about ids anymore 🚀

**Comment by jaimeiniesta** [comment]
👍  LGTM

**Pull request merged**